ibis/backends/pyspark/tests/test_import_export.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+from operator import methodcaller
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    "method",
+    [
+        methodcaller("read_delta", path="test.delta"),
+        methodcaller("read_csv", source_list="test.csv"),
+        methodcaller("read_parquet", path="test.parquet"),
+        methodcaller("read_json", source_list="test.json"),
+    ],
+)
+@pytest.mark.notyet(["pyspark"], raises=NotImplementedError)
+def test_streaming_import_not_implemented(con_streaming, method):
+    method(con_streaming)
```

Comment:
The `notyet`/`notimpl`/... marks shouldn't be used in backend-specific tests, they're more for indicating levels of support for the cross-backend test suite. Instead you should use `pytest.raises` to check that the error is raised appropriately.

```suggestion
def test_streaming_import_not_implemented(con_streaming, method):
    with pytest.raises(NotImplementedError):
        method(con_streaming)
```

Reply:
oh uh okay... [gil had told me to use the `notyet` marker in a flink test](https://github.com/ibis-project/ibis/pull/8751#discussion_r1538137244) - should that have been `pytest.raises`?

Reply:
In that case you moved an `xfail` to a `notyet` mark for testing an error raised *due to an upstream issue*. If/when flink supports that SQL that test will start passing - using a `notyet` marker there (and the reason indicated) lets us know that at some point in the future we may expect that test to start passing. I personally would have been fine with an `xfail` there, but :shrug:.

In contrast, here we're testing that an error we raise in our own code is actually raised. `pytest.raises` is the proper mechanism for testing that, since it explicitly states what's being tested.

Reply:
I see, that makes sense. Thanks for explaining! Just fixed.